src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc

```diff
@@ -58,18 +58,42 @@ using am::MockMessageHelper;
 using ::testing::_;
 using ::testing::Mock;
 using ::testing::Return;
+using ::testing::ReturnNull;
 
 typedef SharedPtr<AddSubMenuRequest> AddSubMenuPtr;
 
 namespace {
 const uint32_t kConnectionKey = 2u;
+const int32_t kMenuId = 5;
 }  // namespace
 
 class AddSubMenuRequestTest
     : public CommandRequestTest<CommandsTestMocks::kIsNice> {
- public:
+ protected:
   AddSubMenuRequestTest()
       : mock_message_helper_(*MockMessageHelper::message_helper_mock()) {}
+  void SetUp() OVERRIDE {
+    app = CreateMockApp();
+    command_msg_ = CreateMessage(smart_objects::SmartType_Map);
+
+    (*command_msg_)[am::strings::msg_params][am::strings::menu_id] = kMenuId;
+    (*command_msg_)[am::strings::params][am::strings::connection_key] =
+        kConnectionKey;
+
+    ON_CALL(app_mngr_, hmi_interfaces())
+        .WillByDefault(ReturnRef(hmi_interfaces_));
+
+    EXPECT_CALL(hmi_interfaces_,
+                GetInterfaceState(am::HmiInterfaces::HMI_INTERFACE_UI))
+        .WillRepeatedly(Return(am::HmiInterfaces::STATE_AVAILABLE));
```

Comment:
@VVeremjova If you don't specify the times this call happens, why don't you change it to `ON_CALL`?